DTOParamConverter.php

```diff
@@ -30,12 +30,14 @@ final class DTOParamConverter implements ParamConverterInterface
     const PROPERTY_OPTION_NULLABLE = 'nullable';
     const PROPERTY_OPTION_DISABLED = 'disabled';
     const PROPERTY_OPTION_OPTIONS = 'options';
+    const PROPERTY_OPTION_UNDEFINEDABLE = 'undefinedable';
 
     const DEFAULT_OPTION_TYPE = 'string';
     const DEFAULT_OPTION_SCOPE = 'request';
     const DEFAULT_OPTION_NULLABLE = true;
     const DEFAULT_OPTION_DISABLED = false;
     const DEFAULT_OPTION_OPTIONS = [];
+    const DEFAULT_OPTION_UNDEFINEDABLE = false;
```

Comment:
Esasen **DEFAULT_OPTION_UNDEFINEDABLE** bu sabit'e ihtiyaç yok. Burada **PROPERTY_OPTION_NULLABLE** kullanılabilir.

Reply:
Zaten var olan nullable Constant'ını kullanmama sebebi onun kullanıldığı yerleri Undefined class'ı ile patlatmak istememesi sanırım.

Reply:
Uygulama içerisinde nullable hiç kullanılmamış dolayısı ile patlatabileceği herhangi bir durum yok. O nedenle nullable kullanmak yeterli olacaktır zaten burada ki amacımız nullable olabilen değerleri handle etmek yeni bir opsiyon tanımlamak kompleks hale getirir. Bir property nullable olarak tanımlandı ise default değeri **Undefined** veya **Null** sınıfları olabilir, eğer kullanıcı bir değer gönderdi ise zaten o değer set edilir (null veya string, integer, vs..).

Reply:
vendor/metglobal/dto-bundle/OptionsResolver/ParameterOptionsResolver.php:11 kısmını incelemenizi tavsiye ederim. Bu sabitlerin her biri burada kullanılıyor

Reply:
Ayrıca nullable / undefined logic'i farklı iki senaryoya hitap ediyorlar. Bir parametre nullable olabilir, hiç gönderilmeyedebilir (undefined).

Reply:
undefined / nullable opsiyonun bir parçası başka bir senaryo değil. undefined nullable opsiyonu durumda kullanılabilir zaten başka bir durumda geçerli değil (kullanılamaz).

Reply:
@Parameter(type="int", nullable=true, undefinedable=true)
{ "abc": 5 } -> 5 (int)
{ "abc": null } -> null
{ } -> Undefined()

@Parameter(type="int", nullable=false, undefinedable=true)
{ "abc": 5 } -> 5 (int)
{ "abc": null } -> 0 (int) çünkü 0 integerlar için default değer.
{ } -> Undefined()

@Parameter(type="int", nullable=false, undefinedable=false)
{ "abc": 5 } -> 5 (int)
{ "abc": null } -> 0 (int)
{ } -> 0 (int)

@Parameter(type="int", nullable=true, undefinedable=false)
{ "abc": 5 } -> 5 (int)
{ "abc": null } -> 0 (int)
{ } -> 0 (int)

Yukarıdaki verileri inceleyerek, farklı veri tiplerine karşı DTOConverter'ın nasıl davrandığını ve nullable ile undefined'ın farkını net olarak görebilirsiniz